### PR TITLE
Feat/recover role page

### PR DESCRIPTION
**Pull request by bitbeckers**
* Adds recover role page
* Implements simple mint hats flow for recovery

### CODE REVIEW

File: app/recover-role/page.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+import { abi as HatsAbi } from "@/lib/hatsAbi";
+
+export default function RecoverRolePage() {
+  const publicClient = usePublicClient();
+  const { data: walletClient, isLoading } = useWalletClient();
+  const { address, isConnected } = useAccount();
+  const [peerAddress, setPeerAddress] = useState("");
+  const [result, setResult] = useState<{ success: boolean; message: string } | null>(null);
+
+  const hatsContractAddress = "0x3bc1A0Ad72417f2d411118085256fC53CBdDd137"; // Sepolia testnet address
```
Comment (pheuberger) [inline app/recover-role/page.tsx:23]: Also something for `constants.ts`

File: app/recover-role/page.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+
+    if (!publicClient) {
+      console.log("PANIC");
+      return;
+    }
+
+    const hatsId = BigInt("0x0000027000020000000000000000000000000000000000000000000000000000"); // https://app.hatsprotocol.xyz/trees/11155111/624?hatId=624.2
```
Comment (pheuberger) [inline app/recover-role/page.tsx:37]: We've used this in the account page too, haven't we? Should also go into `constants.ts`

Comment (bitbeckers) [reply app/recover-role/page.tsx:37]: No this needs to be dynamically infered from the connected user, there's no constant HatsId for all users

File: app/recover-role/page.tsx (hunk 1 above)
Comment (bitbeckers) [reply app/recover-role/page.tsx:23]: let's figure this out when we go cross chain, we might need a handler to support multiple chains

Comment (pheuberger) [reply app/recover-role/page.tsx:23]: SGTM

File: app/recover-role/page.tsx (hunk 2 above)
Comment (pheuberger) [reply app/recover-role/page.tsx:37]: Ah okay, this is just placeholder data?

**Comment by pheuberger** [comment]
The build is failing because this wasn't rebased on top of #8 but that's fine. Once it's merged it will work.

**Review by pheuberger — APPROVED** [review]

**Pull request merged**